### ISSUE TITLE
bug: Linter errors from include files reported in main file

### ISSUE THREAD

**Issue by albertziegenhagel** [fortran-lang/vscode-fortran-support#931]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

If a fortran file `A` includes another file `B`, and the file that is included (`B`) has a syntax error in it, the linter errors (I am using `gfortran`) are incorrectly reported to be in the file `A`.

As an example let file `A` be `prog.f90` with

```f90
! Meaningless comment
! to make the error
! more visible
program prog
  implicit none
  include 'include.inc'

  print*, "Hello Include!"
end program prog
```

and file `B`  is `include.inc` with

```f90

integers :: i

```

(note that there is a type `integers` instead of `integer`).

Here the linter error is reported in `prog.f90` line 2, column 9 but in fact, it is in include.inc.

If I copy the `gfortran` command from the `Modern Fortran` output window to a terminal and invoke gfortran manually it reports

```
include.inc:2:8:

Error: Malformed type-spec at (1)
```

### Screenshots

![image](https://github.com/fortran-lang/vscode-fortran-support/assets/6369416/df01999b-7dbb-4a85-a74a-1be02f0d8385)

### Expected Behaviour

That is actually not that easy to answer. One solution would be to report the errors in the `include.inc` file, but this could lead to problems when the file is included from multiple locations and each inclusion leads to different errors (as an example, one can fix the typo in the `include.inc` file above, but add `-Wunused-variable` to the linter arguments. In this case one will get a warning about an used variable `i` in the example. But if we would include the same file from another subroutine where `i` is used, this will not trigger a warning).

An alternative would be to show all warnings from the includes on the `include 'include.inc'` line in `prog.f90`, but this way one might get a lot of warnings/errors in that single line and it will be hard to figure out where exactly they originate from.

I am open for any other suggestions for solutions to that problem

### Version of Modern Fortran

v3.4.2023062602

### Version of Visual Studio Code

1.79.2

### Platform and Architecture

Windows

**Comment by gnikit** [comment]
Hey @albertziegenhagel thanks for the issue. Can you set the language type of the .inc file to `FreeFormFortran` and see if there's a difference?

I suspect that the file URI for the .inc file is not "registered" via our extension.

Also, gfortran version are you using?

**Comment by albertziegenhagel** [comment]
The language type was already set to `FreeFormFortran`. But even if I rename the file to `include.f90` the error still exists (although in that case I do additionally get the error displayed in the `include.f90` file.

I am running gfortran `13.1.0`:
```
PS C:\> gfortran --version
GNU Fortran (Rev6, Built by MSYS2 project) 13.1.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by gnikit** [comment]
Yes that is correct. The problem is that diagnostics should be created/destroyed when a file is opened/closed.

We can't serve diagnostics to a file that has not been opened yet. In your case, you have opened the .inc file and consequently the error manifests as
"Diagnostics in the wrong file".

If the .inc file had not been opened, I think the expected behaviour would be to show no errors/warnings.
